runtime/runtime/src/lib.rs

```diff
@@ -2084,16 +2084,20 @@ impl Runtime {
 
         let bandwidth_requests = receipt_sink.generate_bandwidth_requests(&state_update, true)?;
 
-        check_balance(
-            &apply_state.config,
-            &state_update,
-            validator_accounts_update,
-            processing_state.incoming_receipts,
-            &promise_yield_result.timeout_receipts,
-            processing_state.transactions,
-            &receipt_sink.outgoing_receipts(),
-            &processing_state.stats,
-        )?;
+        debug_assert_eq!(
+            check_balance(
+                &apply_state.config,
+                &state_update,
+                validator_accounts_update,
+                processing_state.incoming_receipts,
+                &promise_yield_result.timeout_receipts,
+                processing_state.transactions,
+                &receipt_sink.outgoing_receipts(),
+                &processing_state.stats,
+            ),
+            Ok(()),
```

Comment:
Instead of comparing with `Ok(())` should we consider unwrapping the error here for better tracing?

Reply:
`assert_eq!` prints `left` and `right` on a mismatch, see for instance [this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=28eb202c0101fbacc7b747d15aa78b01). Though I might rewrite it to add tracing.